### PR TITLE
docs: add rst file for zigbee shell reference

### PR DESCRIPTION
**Pull request by greg-fer**
Adapted SDK documentation of CLI example to NCS.
KRKNWK-7141 and KRKNWK-6682.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: samples/zigbee/cli/README.rst [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+Zigbee: CLI
+###########
+
+The Zigbee CLI sample demonstrates a Zigbee router (with the possibility of being a coordinator) that uses the Command Line Interface inside Zephyr's :ref:`zephyr:shell_api` for interaction.
+
+The sample can be used for several purposes, including:
+
+* Initial configuration of the network - forming a network as a coordinator, adding devices to the network with the install codes, setting the extended PAN ID
+* Troubleshooting the existing network - reading and writing attributes, subscribing to the reports of the parameters
```
Comment (sebastiandraus) [inline samples/zigbee/cli/README.rst:11]: Not yet implementeed, ZCL commands are to be ported later

File: samples/zigbee/cli/README.rst [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+The CLI can be accessed by using the following interfaces:
+
+* UART
+* RTT
+
+These interfaces are completely independent one from another and can be used simultaneously or exclusively.
+
+The sample demonstrates the usage of commands listed in `Zigbee CLI Reference`_.
+Zigbee CLI is integrated into the system shell accessible over serial connection.
```
Comment (sebastiandraus) [inline samples/zigbee/cli/README.rst:25]: Yes, by default, CLI commands are accessible by UART. Should it mention other interfaces here?

File: samples/zigbee/cli/README.rst [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+   :start-after: set8_start
+   :end-before: set8_end
+
+Use two compatible development kits programmed with this sample.
+Additionally, use either one compatible development kit programmed with :ref:`zigbee_network_coordinator_sample` for testing communication or a third compatible development kits programmed with the Zigbee CLI sample.
+
+User interface
+**************
+
+All the interactions with the application are handled using serial communication.
```
Comment (sebastiandraus) [inline samples/zigbee/cli/README.rst:44]: "can be handled" ? We may use RTT instead, maybe USB CDC later.

File: samples/zigbee/cli/README.rst [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+#. Depending on whether you have access to a standalone Zigbee Coordinator:
+
+   * If you have a standalone Zigbee Coordinator, make sure it is able to accept new nodes.
+   * If you do not have a standalone Zigbee Coordinator, use one of the following solutions:
+
+     - Use a Zigbee Coordinator from the :ref:`zigbee_network_coordinator_sample` sample and flash it on a third board.
+     - Set up one of the CLIs to act as Zigbee Coordinator (see `Setting up CLI device as Zigbee Coordinator`_).
+
+#. By default, the CLI sample implements the Zigbee Router functionality.
+   If you have a coordinator nearby that is listening on channel 16, run the following command on boards programmed with the CLI sample (excluding the coordinator board, if you use a CLI board as coordinator):
```
Comment (sebastiandraus) [inline samples/zigbee/cli/README.rst:77]: "If you have a coordinator nearby using the channel 16"? Justo listening on channel 16 seems not precise.

File: samples/zigbee/cli/README.rst [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  * :file:`zigbee_helpers.h`
+  * :file:`zb_error_handler.h`
+
+* :ref:`dk_buttons_and_leds_readme`
+
+This sample uses the following `nrfxlib`_ libraries:
+
+* :ref:`nrfxlib:zboss`
+
+In addition, it uses the following Zephyr libraries:
```
Comment (sebastiandraus) [inline samples/zigbee/cli/README.rst:149]: Should the `Shell library` be mentioned here?

File: samples/zigbee/cli/README.rst [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+User interface
+**************
+
+All the interactions with the application are handled using serial communication.
+See `Zigbee CLI Reference`_ for the list of available serial commands.
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/zigbee/cli`
```
Comment (sebastiandraus) [inline samples/zigbee/cli/README.rst:50]: It is possible that path to the example will be different, we should remember to eventually change it

File: samples/zigbee/cli/README.rst (hunk 5 above)
Comment (greg-fer) [reply samples/zigbee/cli/README.rst:149]: You tell me -- does the sample use it?

[EDIT: Based on f2f discussion, the sample might use Zigbee Shell library, which will require documentation of its own. @sebastiandraus told me the Zephyr Shell is not used by the sample, so for now no mention of shell will be added.]

File: samples/zigbee/cli/README.rst (hunk 6 above)
Comment (greg-fer) [reply samples/zigbee/cli/README.rst:50]: OK

File: samples/zigbee/cli/README.rst (hunk 5 above)
Comment (greg-fer) [reply samples/zigbee/cli/README.rst:149]: F2F: `Zigbee shell` will be a library that will be added before CLI sample is ready. @tomchy will implement it together with @sebastiandraus .

File: samples/zigbee/cli/README.rst (hunk 2 above)
Comment (greg-fer) [reply samples/zigbee/cli/README.rst]: Discussed f2f.

File: samples/zigbee/cli/README.rst [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+      zdo short
+
+   The command returns the acquired short address of the each CLI node.
+#. Check the EUI64 addresses by running the following command on each board programmed with the CLI sample:
+
+   .. code-block::
+
+      zdo eui64
+
```
Comment (greg-fer) [inline samples/zigbee/cli/README.rst:99]: @sebastiandraus , kindly please provide additional steps and testing result here for checking the network addresses.

File: doc/nrf/ug_zigbee_cli_reference.rst [hunk 8]
```diff
@@ -0,0 +1,620 @@
...
+Send Simple Descriptor Request to the given node and endpoint.
+
+Example:
+
+.. code-block::
+
+   > zdo simple_desc_req 0xefba 10
+   > src_addr=0xEFBA ep=0x260 profile_id=0x0102 app_dev_id=0x0 app_dev_ver=0x5
+     in_clusters=0x0000,0x0003,0x0004,0x0005,0x0006,0x0008,0x0300 out_clusters=0x0300
+   Done
```
Comment (sebastiandraus) [inline doc/nrf/ug_zigbee_cli_reference.rst:288]: Just a general note, using Shell to print command response does not prepend the response with any prompt
```suggestion
   > zdo simple_desc_req 0xefba 10
   src_addr=0xEFBA ep=0x260 profile_id=0x0102 app_dev_id=0x0 app_dev_ver=0x5
   in_clusters=0x0000,0x0003,0x0004,0x0005,0x0006,0x0008,0x0300 out_clusters=0x0300
   Done
```

File: doc/nrf/ug_zigbee_cli_reference.rst [hunk 9]
```diff
@@ -0,0 +1,620 @@
...
+   version
+
+Example:
+
+.. code-block::
+
+   > version
+   CLI: Dec 11 2018 16:14:18
+   ZBOSS: 3.1.0.59
+   Done
```
Comment (sebastiandraus) [inline doc/nrf/ug_zigbee_cli_reference.rst:568]: Just a small update here
```suggestion
   > version
   CLI: Sep  3 2020 13:34:28
   ZBOSS: 3.3.0.2
   Zephyr kernel version: 2.3.99
   Done
```

File: doc/nrf/libraries/zigbee/lib_zigbee_shell.rst [hunk 10]
```diff
@@ -0,0 +1,662 @@
...
+To extend a sample with the Zigbee shell command support, set the following KConfig options:
+
+* ``CONFIG_ZIGBEE_SHELL`` - This option enables Zigbee shell and Zephyr's :ref:`zephyr:shell_api`.
+* ``CONFIG_ZIGBEE_SHELL_ENDPOINT`` - This option specifies the endpoint number to be used by the Zigbee shell instance.
+  Endpoint must be present at the device and you must not register an endpoint handler for this endpoint.
+* ``CONFIG_ZIGBEE_SHELL_DEBUG_CMD`` - This option enables commands useful for testing and debugging.
+
+  .. note::
+    Using debug commands can make the device unstable.
+
```
Comment (greg-fer) [inline doc/nrf/libraries/zigbee/lib_zigbee_shell.rst:30]: @sebastiandraus , shouldn't we add a note here about ``CONFIG_ZIGBEE_SHELL_LOG_LEVEL``? I can imagine people wondering what the relation is between these options.

File: doc/nrf/libraries/zigbee/lib_zigbee_shell.rst [hunk 11]
```diff
@@ -0,0 +1,662 @@
...
+Enabling a device to work with these commands simplifies testing and debugging of an existing network and reduces the amount of reprogramming operations.
+
+.. _zigbee_shell_extending_samples:
+
+Adding support for Zigbee shell commands
+****************************************
+
+The Zigbee shell commands are implemented using Zephyr's :ref:`zephyr:shell_api` interface.
+By default, it uses the UART backend.
+To change this and other Zephyr's shell settings (for example, the prompt or the maximum amount of accepted command arguments), check the documentation page in Zephyr.
```
Comment (greg-fer) [inline doc/nrf/libraries/zigbee/lib_zigbee_shell.rst:16]: @edmont , given your recent changes here:
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_logging.html#shell
Can you have a look at this PR and see if you have any concerns about Zephyr Shell mentions?

**Review by edmont — COMMENTED** [review]
@greg-fer, the statements regarding the usage of Zigbee shell look good to me.

**Review by wbober — APPROVED** [review]
lgtm

**Review by sebastiandraus — APPROVED** [review]

**Pull request merged**